infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -85,32 +130,43 @@ def backup_database_on_interval(seconds):
     )
     while True:
         time.sleep(seconds)
-        backup_database(make_backup_fpath())
+        backup_database(make_object_name())
 
 
 ###################
 # --- Restore --- #
 ###################
 
 
-def get_most_recent_backup_fpath():
+def download_backup_object(session, backup_bucket_name, backup_obj_name, backup_fpath):
+    """Downloads the object_name backup from the backup_bucket_name S3 bucket."""
+    s3_client = session.client("s3")
+    s3_client.download_file(backup_bucket_name, backup_obj_name, backup_fpath)
+
+
+def list_bucket_objects(session, backup_bucket_name):
+    """Returns a list of all objects in the backup_bucket_name S3 bucket."""
+    bucket = session.resource("s3").Bucket(backup_bucket_name)
```

Comment:
This is a type I'd love to have annotated. But if it's hard to find, don't worry about it. `boto3` is _awful_ when it comes to trying to figure out their types.

Reply:
OMG, we should totally look into this library at some point: https://pypi.org/project/boto3-stubs/ It allegedly adds type annotations for lots of the `boto3` functions. The community is not a fan of how typing is (not) managed with `boto3`

Reply:
`def list_bucket_objects(session: boto3.session.Session, backup_bucket_name: str) -> List[str]:`
I have added this already. are you wanting extra documentation of the bucket type for refererence or is the above what you are looking for?

Reply:
resolving with same issue unresolved above

Reply:
For the docstrings can you use the `sphinx` format? You've used the `google` format here. Google is very pretty, but it takes more vertical space. It made more sense to use google before python added type annotations to the language. Now, if you document the type in the docstring AND in the function signature... you've documented the same thing in two places. This makes it easier for the docstring to get out of sync with the actual code.

It should be:

```python
def list_bucket_objects(session: boto3.session.Session, backup_bucket_name: str) -> List[str]:
    """
    One line sentence about what the function does.

    Optional multi-line description that goes into more depth.
    
    :raises DatabaseBackupError: what it takes for this error to be raised (if ever, this is just an example)

    :param session: whatever the session is
    :param backup_bucket_name: whatever this is
    :param something_else: this is a multi-line
        explanation of a parameter
    """
```